### PR TITLE
Siege match export

### PR DESCRIPTION
**Pull request by levysaar**
I've developed a new tool that will use the siege matchup Json,
Therefore I created a new plugin to extract that specific json.

Please let me know if there's a problem, first time opening a pull request :)

### CODE REVIEW

**Comment by levysaar** [comment]
The plugin suppose to generate a json file when entering te siege screen in game.

**Comment by Xzandro** [comment]
Hmm, I'm a bit on the fence about this. Seems a bit too simple to add this to the default plugins. If you need this information, you could also just enable the Full-Logger and be done with it. You just dump the unchange request data to a file, no structional changes or effort to make it more clear.

Thanks, I will think about it a little bit more.

**Comment by Xzandro** [comment]
I understand. The only issue I see with this is that I would need to recompile and release a whole new SWEX version for every small plugin change you wanna make, which is less than ideal over time. Especially for very small / niche sites (as of yet). We might need to figure out a solution for #18 but didn't look into it too much so I don't know if there is an obvious solution to it.

**Comment by levysaar** [comment]
I understand, that's an issue indeed.
a plugin system would be awesome, but I guess it will be hard to implement.

I can create a custom builds (for windows only if I understood correctly) and share the download link, but I feel like it's stealing your hard work, and your credit.

How about implementing a plugin that will use a config file to export basic API commands like the one I need? with no extra logic, just the plain Jsons.
The config file will have the API "key names", separated by new lines/commas,
and when ever one of the API's in the config file is being logged, it will create a file for it.
Basically a full logger filter but with 1 file per API "key name"

A problem I can think of, is the file names for the APIs, but maybe we can use the "key name" by default and concat the wizardID (need to check if all APIs requests has this value)

Please let me know if such thing is possible to create "easily",
and if you would like that kind of thing to exist.

or if not, your thought about me creating custom builds with my plugins and share them.

Thank you!

**Comment by PeteAndersen** [comment]
FWIW I overcame this issue for swarfarm-logger by having some server-side code that will tell the plugin what commands, and data from those commands, to look out for. The plugin then monitors every command and if it matches the "accepted commands" list from the server, it pulls out the specific object keys and ships it off. 


init here: https://github.com/Xzandro/sw-exporter/blob/master/app/plugins/swarfarm-logger.js#L35

processing here: https://github.com/Xzandro/sw-exporter/blob/master/app/plugins/swarfarm-logger.js#L88

**Comment by Xzandro** [comment]
I have two issues with custom core builds.

- Trust: people know that this is the main repo and that it's safe to use. I'm not saying, that you have bad intentions, but can I really be sure, that your build is clear from weird stuff? Probably not.
- Maintaining: to keep SWEX up-to-date it's probably a better idea to keep the core at one place. Otherwise we receive tons of unrelated issues from cluttered custom builds all the time.

I'm not sure that the best solution to all of this really is. Afer a quick google session I saw a few plugin modules, but nothing really that stood out to me and most plugin modules were popular, the demand for this seem rather low.
As I said, I'm not really sure how to proceed with this yet.

**Comment by levysaar** [comment]
@Xzandro I understand the issues, and I would like to keep it on your source repo as well.

What's your idea about a plugin that use a file in app storage (similar to the existing Config.json file)
The internal file will contain a list of api command keys, and the users will be able to edit it.
The plugin logic will be similar to the `accepted_commands` in @PeteAndersen's swarfarm plugin.

The benefit I can see from such plugin, is if someone wants a specific Json for any reason,
he can just add the api command to the file and a plain json output file will be generated
without the need to go to the full log, copy the response and save it to new file yourself.

If you think that kind of plugin is welcome, I can try to develop it, 
might need some help as I'm still new to this.

Other option is I will have a list of `accepted_commands` on my server like swarfarm, 
but then I will need a dedicated plugin, 
and as we know, my web page is still new and small, and might not stand, depends if the community will like it.

**Comment by Xzandro** [comment]
Your idea will still not solve this problem completely sadly. Ideally you can just select a plugin folder, copy plugins in there and SWEX will work with them and provides any events etc. This is not too hard to implement, but the issue is that you can only use 'native' modules with this. As I said, there were plugin modules for this, but not sure which to choose etc. We need to solve #18  Takes a bit of time.

We had file madness in SWProxy, it wasn't great and super confusing.

**Comment by levysaar** [comment]
@Xzandro  
Hi, I changed the plugin to be specific for my site, since I now use more API commands.
Should I close this PR and open a new one or keep this one?

I know you still don't like the idea, but without the plugin available to everyone, my site is just dead.
It's sad to see my hard work without a use because its not easily accessible.

If you prefer, I can have a list of command on my site similar to swarfarm, but in the code there will still be a unique reference to `GetGuildSiegeBattleLog` API I guess, since both attack and defense logs share the same API call. (I might be able to think about something around that)

Please guide me to the direction that will help my site having a plugin without a plugin module in SWExporter (since it will take much more time)

**Comment by Xzandro** [comment]
Hey, good news: this will soon be fixed via the  #212 PR. Where you can finally use external plugins. I'm very satisfied with porksmash's approach with the asar format and we need to test it a bit more and I will release a new version soon if everything runs smoothly.

**Comment by levysaar** [comment]
@Xzandro 
Thank you for the update
and thank you very much @PeteAndersen for your development

if you need another tester, I would like to help testing the plugin system with my site's plugin

**Comment by Xzandro** [comment]
You already can. I merged it into master. We also updated the readme so you can read up how to do it. Spoiler: it's super easy.

**Comment by Xzandro** [comment]
Hi, closing this in favor of the new external plugin support in the versions 0.0.25+. Read up how to create external plugins [here](https://github.com/Xzandro/sw-exporter#developing-plugins).
